### ISSUE TITLE
explicit-module-boundary-types with default values

### ISSUE THREAD

**Issue by DreamTexX** [denoland/deno_lint#1018]
### Lint Name
explicit-module-boundary-types

### Code Snippet

```ts
export function foo(a: string, b: number, c = "default"): void {
  console.log(a, b, c);
}
```

### Expected Result
Linter should run fine

### Actual Result
```
(explicit-module-boundary-types) All arguments should be typed
export function foo(a: string, b: number, c = "default"): void {
                                          ^^^^^^^^^^^^^
    at ***

    hint: Add types to all the function arguments
    help: for further information visit https://lint.deno.land/#explicit-module-boundary-types
```

### Additional Info
Adding a type `c: string = "default"` results in two linter errors: `no-inferrable-types` and `explicit-module-boundary-types`

### Version

deno 1.19.0 (release, x86_64-unknown-linux-gnu)
v8 9.9.115.7
typescript 4.5.2

**Comment by sigmaSd** [comment]
It fails here https://github.com/denoland/deno_lint/blob/main/src/rules/explicit_module_boundary_types.rs#L166

Even if this is an assign, it doesn't have a type because its inferred, but `check_ann` will just trigger the lint since it doesn't see the type and don't know about the type inference I guess?

**Comment by DreamTexX** [comment]
But it also triggers "explicit-module-boundary-types" when specifically adding the type
```ts
export function foo(a: string, b: number, c: string = "default"): void {
  console.log(a, b, c);
}
```

**Comment by Conaclos** [comment]
Specifying the type triggers also a recommended rule [no-inferrable-types](https://lint.deno.land/#no-inferrable-types).

IMO, we should accept undeclared types for all assignment / default param valdiated by `no-inferrable-types`.

**Comment by ah-yu** [comment]
Agree. I think all params with default values should be ignored in this rule like eslint does.

See https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/explicit-module-boundary-types.ts#L206-L207

**Comment by bartlomieju** [comment]
@ah-yu sounds good to me

**Comment by DreamTexX** [comment]
Hey,
I still get an error when adding a type to a default parameter

```
(explicit-module-boundary-types) All arguments should be typed
    propertyKey: symbol | string = 'prototype',
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    at /home/.../metadata.ts:77:4

    hint: Add types to all the function arguments
    help: for further information visit https://lint.deno.land/#explicit-module-boundary-types
```

**Comment by ah-yu** [comment]
@DreamTexX are you using the latest version of deno? If not, can you upgrade deno and try again?
